src/OpenTelemetry.Api/Metrics/LabelSet.cs

```diff
@@ -21,73 +21,17 @@ namespace OpenTelemetry.Metrics
 {
     /// <summary>
     /// Normalized name value pairs of metric labels.
-    /// TODO: Most of the logic here would be moved into SDK from the API.
     /// </summary>
-    public class LabelSet
+    public abstract class LabelSet
     {
         /// <summary>
         /// Empty LabelSet.
         /// </summary>
-        public static readonly LabelSet BlankLabelSet = new LabelSet();
+        public static readonly LabelSet BlankLabelSet = new BlankLabelSet();
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="LabelSet"/> class.
+        /// Gets or sets the labels after sorting and removing duplicates.
         /// </summary>
-        /// <param name="labels">labels from which labelset should be constructed.</param>
-        public LabelSet(IEnumerable<KeyValuePair<string, string>> labels)
-        {
-            this.Labels = labels;
-            this.LabelSetEncoded = this.GetLabelSetEncoded(labels);
-        }
-
-        private LabelSet()
-        {
-        }
-
-        /// <summary>
-        /// Gets or sets the labels for this LabelSet.
-        /// </summary>
-        public IEnumerable<KeyValuePair<string, string>> Labels { get; set; }
-
-        /// <summary>
-        /// Gets or sets the label set in encoded form for this LabelSet.
-        /// </summary>
-        public string LabelSetEncoded { get; set; }
-
-        /// <inheritdoc/>
-        public override int GetHashCode()
-        {
-            return this.LabelSetEncoded.GetHashCode();
-        }
-
-        /// <inheritdoc/>
-        public override bool Equals(object obj)
-        {
-            return this.LabelSetEncoded.Equals(((LabelSet)obj).LabelSetEncoded);
-        }
-
-        private string GetLabelSetEncoded(IEnumerable<KeyValuePair<string, string>> labels)
-        {
-            // Sort using keys.
-            var ordered = this.Labels.OrderBy(x => x.Key);
-            StringBuilder encoder = new StringBuilder();
-            bool isFirstLabel = true;
-
-            // simple encoding.
-            foreach (KeyValuePair<string, string> label in ordered)
-            {
-                if (!isFirstLabel)
-                {
-                    encoder.Append("\0");
-                }
-
-                encoder.Append(label.Key);
-                encoder.Append("\t");
-                encoder.Append(label.Value);
-                isFirstLabel = false;
-            }
-
-            return encoder.ToString();
-        }
+        public abstract IEnumerable<KeyValuePair<string, string>> Labels { get; set; }
```

Comment:
Would it make sense to leave an instance here?

```suggestion
        public abstract IEnumerable<KeyValuePair<string, string>> Labels { get; set; } = Enumerable.Empty<KeyValuePair<string, string>>();
```